docs/contributing/999-rfcs/2024-06-02-ts-client.md

```diff
@@ -0,0 +1,354 @@
+    ],
+  },
+);
+
+new aws.iam.RolePolicyAttachment(
+  name: "AcmeClientPolicyAttachment",
+  policyArn: policy.arn,
+  role: role.name,
+);
+
+new k8s.ServiceAccount(
+  name: "acme-client",
+  metadata: {
+    name: "acme-client",
+  },
+  annotations: {
+    "eks.amazonaws.com/role-arn": role.arn,
+  },
+);
+```
+
+ConfigMaps are a standard mechanism for storing configuration data so it can be used to store the context data.
+
+```wing
+new k8s.ConfigMap(
+  name: "client-ctx",
+  data: {
+    "ctx": client.context,
+  },
+);
+```
+
+Otherwise, the data can be mapped similarly to the other methods like SSM or buckets.
+If you're using terraform/cfn, an output can be created with the context be managed manually after deployment.
+
+```wing
+// cdktf
+new cdktf.Output(
+  name: "ClientContext",
+  value: client.context,
+);
+
+/// aws-cdk-lib
+new awscdk.CFNOutput(
+  name: "ClientContext",
+  value: client.context,
+);
+```
+
+This finishes the setup of the client and the responsibilities of the platform team, so we now switch over to the application developers.
+They can define a pod with the client context and service account in the namespace they've been alloted:
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: app
+  namespace: app
+  spec:
+    serviceAccountName: acme-client
+    containers:
+      - name: app
+        image: app-image
+        env:
+          - name: WING_CONTEXT
+            valueFrom:
+              configMapKeyRef:
+                name: ctx
+                key: ctx
+```
+
+And now `app-image` can use the client to interact with the bucket!
+
+To run locally, the platform team can provide a script to run the core infrastructure (`wing it`) run a 
+local cluster with all the same expected config maps and service accounts to just work.
+
+### AWS Lambda
+
+This walks through the setup of a client by a central platform team for application teams deploying to AWS Lambda.
+
+```wing
+bring cloud;
+bring clients;
+bring util;
+bring "@cdktf/provider-aws" as aws;
+bring "cdktf" as cdktf;
+
+let bucket = new cloud.Bucket() as "shared";
+
+let client = new typescript.Client(
+  name: "@acme/infra-client",
+  lifts: @lift({ bucket: [put] }),
+);
+
+// Set up an SSM parameter to store the context
+new aws.ssmParameter.SsmParameter(
+  name: "/acme/infra-client/context",
+  type: "String",
+  value: client.context,
+);
+```
+
+To extract the necessary permissions, we set up the wing.toml file to create a new policy:
+
+```toml
+[tf-aws]
+client_policy = "AcmeClientPolicy"
+```
+
+Now the application developers need to setup the lambda with two steps:
+
+1. Map the SSM parameter as an environment variable, which is a directly supported feature in AWS
+2. Attach the created policy to the lambda role
+
+For local development, the owners of the central infrastructure can provide a script to lookup the SSM parameter and add the environment variable to the lambda either via an env or whatever deployment process being used.
```

Comment:
One way to make sure we got this design right is to refactor `cloud.Function` (and any other compute resources) to use this layer under the hood. Eventually this should converge into a single layer that allows runtime code to interact with Wing resources.